src/basic/fd-util.c

```diff
@@ -822,6 +822,25 @@ int fd_reopen(int fd, int flags) {
         return new_fd;
 }
 
+int fd_reopen_propagate_flags(int fd, int flags, int propagate_flags) {
+        /* Invokes fd_reopen(fd, flags),
+         * but propagates selected flags if set on the original fd.
+         *
+         * You should use this for example if the original fd potentially is O_APPEND,
+         * otherwise we get rather "unexpected" behavior.
+         * Unless you intentionally want to overwrite pre-existing data,
+         * and have your output overwritten by the next guy.
```

Comment:
we break lines at 109ch btw, see .editorconfig (and emacs/vim counterparts)

Reply:
Okay ;-)
Though I actually did a quick histogram first, and even though there is a clear drop of frequency around 110, I though I was okay still.
```
find . -name "*.c" | xargs perl -nle 'print(int((length($_)+9)/10)*10)' | sort -n | uniq -c
...
  18205 100
  17229 110
   8644 120
   3852 130
   2369 140
   1652 150
    986 160
    852 170
    852 180
    435 190
    175 200
...
      6 560
      1 940
      2 950
```
 :-D

Reply:
oh, our coding style has changed in the past, and we typically do not re-break everything just because we change the rules. But we do expect that new code follows the rules.

(and this rule isn't the most important anyway, we ignore it every time and then...)

Reply:
the rules are written down in here in prose btw:

https://systemd.io/CODING_STYLE/#formatting